core/tauri/src/image/plugin.rs

```diff
@@ -5,46 +5,41 @@
 use crate::plugin::{Builder, TauriPlugin};
 use crate::{command, AppHandle, Image, Manager, ResourceId, Runtime};
 
+use std::sync::Arc;
+
 use super::JsImage;
 
 #[command(root = "crate")]
-fn new<R: Runtime>(app: AppHandle<R>, image: JsImage<'_>) -> crate::Result<ResourceId> {
-  let image: Image<'_> = image.try_into()?;
-  let image = image.to_owned();
-  let mut resources_table = app.resources_table();
-  let rid = resources_table.add(image);
-  Ok(rid)
-}
-#[command(root = "crate")]
-fn from_png_bytes<R: Runtime>(app: AppHandle<R>, image: JsImage<'_>) -> crate::Result<ResourceId> {
-  let image: Image<'_> = image.try_into()?;
-  let image = image.to_owned();
+fn new<R: Runtime>(
+  app: AppHandle<R>,
+  rgba: Vec<u8>,
+  width: u32,
+  height: u32,
+) -> crate::Result<ResourceId> {
+  let image = JsImage::Rgba {
+    rgba: &rgba,
+    width,
+    height,
+  }
+  .into_img(&app)?;
   let mut resources_table = app.resources_table();
-  let rid = resources_table.add(image);
-  Ok(rid)
-}
-#[command(root = "crate")]
-fn from_ico_bytes<R: Runtime>(app: AppHandle<R>, image: JsImage<'_>) -> crate::Result<ResourceId> {
-  let image: Image<'_> = image.try_into()?;
-  let image = image.to_owned();
-  let mut resources_table = app.resources_table();
-  let rid = resources_table.add(image);
+  let rid = resources_table.add(Arc::try_unwrap(image).unwrap().to_owned());
```

Comment:
I guess we can just use `Image::new(&rgba, width, height).to_owned()` instead or wrapping in an Arc and unwraping